packages/eas-cli/src/commands/release/publish.ts

```diff
@@ -0,0 +1,152 @@
+      required: false,
+    }),
+    publishMessage: flags.string({
+      description: 'Short message describing the updates.',
+      required: false,
+    }),
+    json: flags.boolean({
+      description: `return a json with the edited release's ID and name.`,
+      default: false,
+    }),
+  };
+
+  async run() {
+    let {
+      flags: { json: jsonFlag, releaseName, runtimeVersion, publishMessage, inputDir },
+    } = this.parse(ReleasePublish);
+
+    const projectDir = await findProjectRootAsync(process.cwd());
+    if (!projectDir) {
+      throw new Error('Please run this command inside a project directory.');
+    }
+    const accountName = await getProjectAccountNameAsync(projectDir);
+    const {
+      exp: { slug },
+    } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+    const projectId = await ensureProjectExistsAsync({
+      accountName,
+      projectName: slug,
+    });
+
+    if (!inputDir) {
+      const validationMessage = 'input directory must be specified.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ inputDir } = await promptAsync({
+        type: 'text',
+        name: 'inputDir',
+        message: 'Please enter the name of the directory created by the bundler:',
+        initial: 'dist',
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    if (!releaseName) {
+      const validationMessage = 'release name may not be empty.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ releaseName } = await promptAsync({
+        type: 'text',
+        name: 'releaseName',
+        message: 'Please enter the name of the release to publish on:',
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    if (!runtimeVersion) {
+      const validationMessage = 'runtimeVersion name may not be empty.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ runtimeVersion } = await promptAsync({
+        type: 'text',
+        name: 'runtimeVersion',
+        message: 'Please enter the runtime version of the updates',
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    if (!publishMessage) {
+      const validationMessage = 'publish message may not be empty.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ publishMessage } = await promptAsync({
+        type: 'text',
+        name: 'publishMessage',
+        message: `Please enter a publication message.`,
+        initial: (await getLastCommitMessageAsync())?.trim(),
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    const { id: releaseId } = await getReleaseByNameAsync({
+      appId: projectId,
+      releaseName: releaseName!,
+    });
+
+    log.withTick(
+      `️Publishing "${publishMessage}" to ${releaseName}@${runtimeVersion} to on project ${chalk.bold(
```

Comment:
Not sure what kind of messages people write, but if they're anything like commit messages (ranging from short to long), inlining the message here might not look great.

Because there are so many fields, maybe consider printing a differently formatted summary instead? e.g.
```
------
message             Hello world
release             1.0.0
runtime version     49
------
Publishing...
```

Reply:
How does this look?

<img width="695" alt="Screen Shot 2020-12-22 at 7 47 06 PM" src="https://user-images.githubusercontent.com/1220444/102957122-7a219c80-448e-11eb-82d8-50a834cebcb7.png">

Reply:
Looks great! Maybe add an empty line before the summary?